### ISSUE TITLE
Mismatch fused softmax kernel calcualtion results

### ISSUE THREAD

**Issue by yidong72** [NVIDIA/apex#1493]
I found there is a mismatch between fused softmax kernel calculation vs pytorch softmax calculation.

I use the pytorch:22.07 container or the latest apex master branch. Here is how to reproduce it:

```python
import torch
from apex.transformer.functional.fused_softmax import FusedScaleMaskSoftmax
from apex.transformer.enums import AttnMaskType
from nemo.collections.nlp.modules.common.megatron.utils import attention_mask_func

scale_mask_softmax = FusedScaleMaskSoftmax(
    True,
    False,
    AttnMaskType.padding,
    True,
    attention_mask_func,
    True,
    1.0,
)

att_score = torch.load('./attention_score.pth')
att_mask = torch.load('attention_mask.pth')


fused_kernel_result = scale_mask_softmax.forward_fused_softmax(att_score, att_mask)

torch_result = scale_mask_softmax.forward_torch_softmax(att_score, att_mask)
#numbers do not match
error = (fused_kernel_result - torch_result).abs().max()
print(error)

```
where you can download the serialized `attention_score.pth` and 'attention_mask.pth' files at
[attention_mask_tensors.zip](https://github.com/NVIDIA/apex/files/9635041/attention_mask.zip)

This is causing poor model inference quality. Could you help to take a look thanks.
